### ISSUE TITLE
[Failing Test] [sig-network] DNS HostNetwork spec.Hostname field is silently ignored and the node hostname is used when hostNetwork is set to true for a Pod 

### ISSUE THREAD

**Issue by saschagrunert** [cri-o/cri-o#8167]
### Which jobs are failing?

- ci/prow/ci-rhel-e2e
- ci/prow/ci-e2e
- ci/prow/ci-e2e-evented-pleg
- ci/prow/ci-cgroupv2-e2e-crun
- ci/prow/ci-crun-e2e
- ci/prow/ci-e2e-conmonrs
- ci/prow/ci-cgroupv2-e2e

### Which tests are failing?

`[sig-network] DNS HostNetwork spec.Hostname field is silently ignored and the node hostname is used when hostNetwork is set to true for a Pod `

### Since when has it been failing?

A couple of days now.

### Reason for failure (if possible)

```
{ failed [FAILED] expected hostname: 10.0.0.2, got: ci-op-ngp0fq79-3f73e-buildhost
In [It] at: k8s.io/kubernetes/test/e2e/network/dns.go:690 @ 05/13/24 10:09:04.27

There were additional failures detected after the initial failure. These are visible in the timeline
}
```

### Anything else we need to know?

- Refers to
  - https://github.com/kubernetes/kubernetes/issues/67019
  - https://github.com/kubernetes/kubernetes/pull/124428
- I'm able to reproduce it locally

**Comment by saschagrunert** [comment]
I assume this is test-setup related somehow. Not sure if that's the cause: https://github.com/cri-o/cri-o/blob/64fbde3a89741ca4fd64353e43cbf49f36c35a76/contrib/test/ci/system.yml#L30-L38

**Comment by saschagrunert** [comment]
The fix got merged but we're still testing with kubelet `v1.31.0-alpha.0.689+db82fd1604ebf3` in CI.

**Comment by saschagrunert** [comment]
Done, CI got updated.
